### PR TITLE
Standard tools template

### PR DESCRIPTION
**Pull request by smarterclayton**
Testing with image-registry.  Workflow is:

1. bootstrap your repo using the update script
2. do your customization
3. later on, run update again to get the latest code and refactor your scripts

Future changes might gitsubmodule this more cleanly, but for now we're going to do the quick and dirty.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Let me know when you want a review and what review you would like

**Comment by smarterclayton** [comment]
Review please:

1. conventions in makefile
2. update script
3. subset of tools and hack/ included
4. how I moved things to constants.sh (that's the only file most people should change)

**Comment by smarterclayton** [comment]
Note that openshift/image-registry#6 is now in sync with this

File: tools/hack/golang/hack/build-rpms.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+if [[ -n "${dirty}" && "${OS_GIT_TREE_STATE}" == "dirty" ]]; then
+	os::log::warning "Repository is not clean, performing fast build and reusing _output"
+
+	# build and output from source to destination
+	mkdir -p "${rpm_tmp_dir}"
+	ln -fs "${OS_ROOT}" "${rpm_tmp_dir}/SOURCES"
+	ln -fs "${OS_ROOT}" "${rpm_tmp_dir}/BUILD"
+
+	rpmbuild -bb "${OS_RPM_SPECFILE}" \
```
Comment (stevekuznetsov) [inline tools/hack/golang/hack/build-rpms.sh]: nice

File: tools/hack/golang/hack/lib/OWNERS [hunk 2]
```diff
@@ -0,0 +1,2 @@
+reviewers:
```
Comment (stevekuznetsov) [inline tools/hack/golang/hack/lib/OWNERS]: omit the file if you don't want to have me in there

File: tools/hack/golang/hack/lib/constants.sh [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+readonly OS_TEST_TARGETS=( )
+
+readonly OS_ALL_IMAGES=(
+  openshift/origin-docker-registry
+)
+
+# os::build::get_product_vars exports variables that we expect to change
+# depending on the distribution of Origin
+function os::build::get_product_vars() {
```
Comment (stevekuznetsov) [inline tools/hack/golang/hack/lib/constants.sh]: "constants"

File: tools/hack/golang/hack/lib/OWNERS (hunk 2 above)
Comment (smarterclayton) [reply tools/hack/golang/hack/lib/OWNERS]: Hrm, I think i need to take OWNERS out of this dir - we should be at the top level.

File: tools/hack/golang/hack/lib/constants.sh (hunk 3 above)
Comment (smarterclayton) [reply tools/hack/golang/hack/lib/constants.sh]: You want to rename the function?

**Comment by smarterclayton** [comment]
> How does this handle openshift/origin-aggregated-logging@b816a2b#diff-500746274e2e9471f21c105f5d0a91a6 ?

Doesn't, Jordan and I are going to have to talk about that.  oc is getting split out, but we need a way to ratchet the dependency tree somehow.

File: tools/hack/golang/hack/lib/constants.sh (hunk 3 above)
Comment (stevekuznetsov) [reply tools/hack/golang/hack/lib/constants.sh]: no, is a dumb comment is all :)

**Pull request merged**